core/src/main/java/pl/skidam/automodpack_core/protocol/DownloadClient.java

```diff
@@ -192,7 +192,7 @@ public PreValidationConnection(InetSocketAddress address, InetSocketAddress mine
         SSLContext context = createSSLContext(keyStore, interceptedCertificateChain::set);
         SSLSocketFactory factory = context.getSocketFactory();
         // The createSocket(Socket, host, port, autoClose) wraps the existing plain socket.
-        SSLSocket sslSocket = (SSLSocket) factory.createSocket(plainSocket, address.getHostName(), address.getPort(), true);
+        SSLSocket sslSocket = (SSLSocket) factory.createSocket(plainSocket, modpackAddresses.hostAddress.getHostName(), modpackAddresses.hostAddress.getPort(), true);
```

Comment:
`modpackAddresses.hostAddress.getHostName()` can trigger a name lookup if `modpackAddresses.hostAddress` was created with an IP address. This can lead to connecting to a different server than intended by the user / Minecraft server owner. So, unless this is specifically needed, we probably want to use `getHostString()` instead.